### PR TITLE
Permissions component added 

### PR DESCRIPTION
**Pull request by chinmehta**
[x] Wrapper component added to add content when partner preview config is true.
[x] Custom hook added which return partner preview config.
[x] Reference in homepage, classicAssociationTable, index updated.
[x] Part of Issue: https://github.com/opentargets/issues/issues/2714

### CODE REVIEW

File: apps/platform/src/hooks/usePermissions.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import config from '../config';
+
+function usePermissions() {
+  return config.profile.isPartnerPreview;
+}
```
Comment (carcruz) [inline apps/platform/src/hooks/usePermissions.js:5]: Please expose a variable from the hook to make it more clear

Change:
```
isPartnerPreview = usePermissions()
```
to:

```
const { isPartnerPreview } = usePermissions()
```

File: apps/platform/src/components/PrivateWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import usePermissions from '../hooks/usePermissions';
+
+function PrivateWrapper (props) {
+  const isPartnerPreview = usePermissions();
+
+  if (isPartnerPreview) return <>{props.children}</>;
+  else return <></>;
```
Comment (carcruz) [inline apps/platform/src/components/PrivateWrapper.jsx:8]: Unnecesary `else` before a `return`

**Review by carcruz — CHANGES_REQUESTED** [review]

File: apps/platform/src/hooks/usePermissions.js (hunk 1 above)
Comment (chinmehta) [reply apps/platform/src/hooks/usePermissions.js]: done

File: apps/platform/src/components/PrivateWrapper.jsx (hunk 2 above)
Comment (chinmehta) [reply apps/platform/src/components/PrivateWrapper.jsx]: done

**Review by carcruz — APPROVED** [review]

**Pull request merged**